docs/Makefile

```diff
@@ -21,26 +21,36 @@ endif
 
 all: deps api build
 
-api:
-	$(QUARTODOC) build
-	$(QUARTODOC) interlinks
+ensure-dev:
+	$(MAKE) -C .. dev
+
+api: ensure-dev
+	@echo "::group::quartodoc interlinks"
+	$(UV) run $(QUARTODOC) interlinks
+	@echo "::endgroup::"
+	@echo "::group::quartodoc build"
+	$(UV) run $(QUARTODOC) build --verbose
+	@echo "::endgroup::"
 	cp -r _extensions/ reference/_extensions  # Required to render footer
 
-build:
-	CURRENT_YEAR=$(CURRENT_YEAR) \
-	PROJECT_VERSION=$(PROJECT_VERSION) \
-	$(QUARTO) render
+build: ensure-dev
+	source ../.venv/bin/activate && \
+		CURRENT_YEAR=$(CURRENT_YEAR) \
+		PROJECT_VERSION=$(PROJECT_VERSION) \
+		$(QUARTO) render
 
 clean:
 	rm -rf _extensions _inv _site .quarto reference objects.json
 	find . -type d -empty -delete
 
-deps:
-	$(UV) pip install --upgrade pip -r requirements-site.txt
+_extensions/posit-dev/posit-docs/_extension.yml:
 	$(QUARTO) add --no-prompt posit-dev/product-doc-theme@v4.0.2
+_extensions/machow/interlinks/_extension.yml:
```

Comment:
Can we still enforce upgrades to Quartodoc this way?

Reply:
No. Currently docs are not kept in version control. So GHA should pass as it is a clean environment.

I'm up for adding the quarto extensions folder though. This would resolve the issue and isn't much in files.

Reply:
Makes sense. Actually, since `make clean` clears out the extensions folder, it seems reasonable. My concern was for local development where these files are long lived and become out of date, but `make clean` is a decent escape hatch.